quickwit-search/src/collector.rs

```diff
@@ -320,13 +326,13 @@ fn merge_leaf_responses(
         .map(|leaf_response| leaf_response.num_hits)
         .sum();
     let failed_splits = leaf_responses
-        .iter()
-        .flat_map(|leaf_response| leaf_response.failed_splits.iter().cloned())
+        .iter_mut()
+        .flat_map(|leaf_response| std::mem::take(&mut leaf_response.failed_splits))
```

Comment:
No need to cloned each `failed_splits` reported

Reply:
You are correct, but the previous code is simpler to read.

The clone you are trying to avoid is tiny and only happens if we have failed splits. This is not something we need to optimize. "Premature optimization is the root of all evil."

Reply:
You are right. Let me revert and make it a bit easier to read